### ISSUE TITLE
 ignore the string starts with underscore.

### ISSUE THREAD

**Issue by manaphp** [klesun/deep-assoc-completion#169]
can we ignore the string start with underscore.?

![QQ截图20201119113246](https://user-images.githubusercontent.com/16060165/99631415-891fc400-2a76-11eb-8c27-9f8df9c8b47b.png)

**Comment by manaphp** [comment]
https://github.com/klesun/deep-assoc-completion/issues/143#issue-554552548

**Comment by klesun** [comment]
Hi.
In general case no, but maybe some annotation like `@noDeepAssoc` or something could be added on the properties you want to exclude, or something.

**Comment by klesun** [comment]
Actually, are `_snapshot` and `_last_refresh` `private` fields per chance, while others are `protected`? I take it you get this completion from `get_object_vars()` in the scope where these private fields are not accessible, maybe that could be considered in the plugin...

**Comment by klesun** [comment]
![image](https://user-images.githubusercontent.com/5202330/99652174-2b807d00-2a60-11eb-8408-7e0aa2f37d7c.png)

**Comment by manaphp** [comment]
i change from protected to private, but it no effect...

**Comment by manaphp** [comment]
> Hi.
> In general case no, but maybe some annotation like `@noDeepAssoc` or something could be added on the properties you want to exclude, or something.

supported now?!

**Comment by klesun** [comment]
No, not supported, would need to implement, I'm just throwing options of how it _could_ be done.

**Comment by manaphp** [comment]
> No, not supported, would need to implement, I'm just throwing options of how it _could_ be done.

array_diff_assoc(get_class_vars(new static),['_last_refresh'=>1])


don't work..

**Comment by klesun** [comment]
Sorry, realistically speaking, I'll probably won't have time to implement this feature, since I've even been ignoring bugs and typing issues for months. I do intend to dedicate small part of my time to deep-assoc, but it will be barely enough for others, more impactful issues.

**Comment by klesun** [comment]
Actually, wasn't it same thing as https://github.com/klesun/deep-assoc-completion/issues/120 ?
